pennylane/decomposition/resources.py

```diff
@@ -350,7 +361,7 @@ def pow_resource_rep(base_class, base_params, z):
         z (int or float): the power
 
     """
-    if not isinstance(z, int) or z < 0:
+    if not qml.math.is_abstract(z) and (not _is_integer(z) or z < 0):
```

Comment:
```suggestion
    if (not qml.math.is_abstract(z)) and (not _is_integer(z) or z < 0):
```
Wish formatter doesn't complain